core-primitives/top-pool/src/validated_pool.rs

```diff
@@ -197,11 +197,14 @@ where
 	) -> Result<ExtrinsicHash<B>, B::Error> {
 		match tx {
 			ValidatedOperation::Valid(tx) => {
-				let imported = self.pool.write().unwrap().import(tx, shard)?;
+				let imported =
+					self.pool.write().map_err(|_| error::Error::UnlockError)?.import(tx, shard)?;
```

Comment:
These `unwrap()`s fortunately were not very problematic - they could only happen when the lock is poisoned (in which case we have a problem anyway and cannot continue). But explicitly handling that error is cleaner 👍